blocks/api/parser.js

```diff
@@ -45,31 +44,150 @@ export function getBlockAttributes( blockNode, blockSettings ) {
 	return attrs;
 }
 
+function htmlEscape( str ) {
+	return str
+		.replace( /&/g, '&amp;' )
+		.replace( /"/g, '&quot;' )
+		.replace( /'/g, '&#39;' )
+		.replace( /</g, '&lt;' )
+		.replace( />/g, '&gt;' );
+}
+
+function htmlUnescape( str ) {
+	return str
+		.replace( /&quot;/g, '"' )
+		.replace( /&#39;/g, '\'' )
+		.replace( /&lt;/g, '<' )
+		.replace( /&gt;/g, '>' )
+		.replace( /&amp;/g, '&' );
+}
+
 /**
  * Returns a list of blocks extracted from the Post Content
  *
  * @param  {String} content The post content
  * @return {Array}          Block list
  */
 export default function parse( content ) {
-	return grammarParse( content ).reduce( ( memo, blockNode ) => {
-		// Use type from block node, otherwise find unknown handler
-		let { blockType = getUnknownTypeHandler() } = blockNode;
-
-		// Try finding settings for known block type, else again fall back
-		let settings = getBlockSettings( blockType );
-		if ( ! settings ) {
-			blockType = getUnknownTypeHandler();
-			settings = getBlockSettings( blockType );
+	// First, convert comment delimiters into temporary <wp-block> "tags" so
+	// that TinyMCE can parse them.  Examples:
+	//   In  : <!-- wp:core/text -->
+	//   Out : <wp-block slug="core/text">
+	//   In  : <!-- /wp:core/text -->
+	//   Out : </wp-block>
+	//   In  : <!-- wp:core/embed url:youtube.com/xxx& -->
+	//   Out : <wp-block slug="core/embed" attributes="url:youtube.com/xxx&amp;">
+	content = content.replace(
+		/<!--\s*(\/?)wp:([a-z0-9/-]+)((?:\s+[a-z0-9_-]+:[^\s]+)*)\s*-->/g,
+		function( match, closingSlash, slug, attributes ) {
+			if ( closingSlash ) {
+				return '</wp-block>';
+			}
+
+			if ( attributes ) {
+				attributes = ' attributes="' + htmlEscape( attributes.trim() ) + '"';
+			}
+			return '<wp-block slug="' + slug + '"' + attributes + '>';
+		}
+	);
+
+	// Create a custom HTML schema
+	const schema = new tinymce.html.Schema();
```

Comment:
>I assumed the idea behind referencing the global directly in Editable was that as a plugin we wanted to use the TinyMCE version that's bundled with WordPress.

Right, and we can do that with [Webpack externals](https://webpack.js.org/configuration/externals/#components/sidebar/sidebar.jsx). I'm fine to leave this for a separate pull request.